### ISSUE TITLE
Segmentation fault on onInMeeting_() call

### ISSUE THREAD

**Issue by FStupar** [zoom/meetingsdk-linux-raw-recording-sample#8]
Used Ubuntu Dockerfile,  
Built with newest SDK, added necessary tokens and information. 
The bot joins the meeting, but as soon as it's admitted it throws a segmentation fault.
Managed to narrow it down to the onInMeeting_() call that happens in
MeetingServiceEventListener::onMeetingStatusChanged once the status is MEETING_STATUS_INMEETING

Any help would be appreciated.

**Comment by tanchunsiong** [comment]
@FStupar you are on 5.17.11?

**Comment by FStupar** [comment]
@tanchunsiong yes, latest linux build, 5.17.11

**Comment by FStupar** [comment]
@tanchunsiong  
Yep, Dockerfile dor Ubuntu, not changed in any way, fully populated config.txt, correct meeting number password and jwt. 

The bot manages to connect to the meeting just fine, the segmentation fault happens right after that, when calling the onInMeeting_ from the MeetingServiceEventListener

to be precise, segmentation fault occurs right when the onInMeeting_ is called.

The host was originally macOS then we migrated it to gitlab so to remove any possibility of the problem being the host.

**Comment by tanchunsiong** [comment]
@FStupar has the code been changed?

onInMeeting_  invokes back a method in the main thread. If you are missing the method, or if it is nullptr, you might get the seg fault.

**Comment by uro-sh** [comment]
@tanchunsiong I had the same problem, and by some additional debugging, I have figured out the segmentation fault happens on the line 137:
https://github.com/zoom/meetingsdk-linux-raw-recording-sample/blob/77cd735134b6690cc7a83829ffc2f6d1ed6ed92f/demo/meeting_sdk_demo.cpp#L137
Even when I comment the line, and assume the recording rights are there, I got the segmentation fault on the line 140:
https://github.com/zoom/meetingsdk-linux-raw-recording-sample/blob/77cd735134b6690cc7a83829ffc2f6d1ed6ed92f/demo/meeting_sdk_demo.cpp#L140

Do you have any idea why those SDK methods cannot be invoked?
Also, one more nuance: my config.txt file do not contain recording_token. Is it mandatory for basic recording or not? (I assumed not according you the readme file provided).

**Comment by uro-sh** [comment]
@tanchunsiong Ok, with the latest version of SDK (just released), the seg fault is not there anymore, but I am having the SDKError 12 ("Cannot start raw recording: no permissions yet, need host, co-host, or recording privilege") even though I have allowed all participants to start recording local files (see image attached).

![Screenshot 2024-04-18 at 13 06 33](https://github.com/zoom/meetingsdk-linux-raw-recording-sample/assets/156791923/f4a2335c-c1c6-4733-b0cd-4a71bac99e5b)

Do I now need to generate "recording_token" for that?

**Comment by tanchunsiong** [comment]
@uro-sh that error is a bit odd, possibly that m_pMeetingService->GetMeetingRecordingController() is throwing nullptr

For the no permission issue, the host can be prompted to provide permission (recommended way).
Recording token requires you to get permission the admin of the meeting which you are joining, to fetch the token prior to joining the meeting.

This is the specific token which is necessary
https://developers.zoom.us/docs/api/rest/reference/zoom-api/methods/#operation/meetingLocalRecordingJoinToken

**Comment by uro-sh** [comment]
@tanchunsiong It is not throwing nullptr, I have checked it. Also, I am able to quickly after the bot joins the meeting click on Participants -> Select the bot -> Click More... -> Select "Allow local recording", and the recording process started and was successful (the PCM file was generated). But without allowing that option manually the bot cannot start recording.

Is there a code example for prompting for the permission (the recommended way)?

**Comment by uro-sh** [comment]
@tanchunsiong Thank you, that worked for me!
